pkg/kubelet/cm/devicemanager/endpoint.go

```diff
@@ -87,6 +91,14 @@ func (e *endpointImpl) callback(resourceName string, devices []pluginapi.Device)
 	e.cb(resourceName, devices)
 }
 
+func (e *endpointImpl) getDevices(resourceName string) []pluginapi.Device {
```

Comment:
Could you use the m.healthyDevices map in the manager rather than creating this map here?

Having the endpoint maintain its own map of Devices was what we used to do in the past but it created a lot of complexity.

Currently, the manager maintains its own map of Devices through populating `m.healthyDevices` and `m.unhealthyDevices` could we rewrite these maps in a way that would work for this use case?

One idea could be to change the type of these maps to pluginapi.Device and generate the sets in the GetCapacity function.

What do you think?

Reply:
We can't use healthyDevices as it as because it doesn't contain the topology information. What you suggested is totally plausible, or if we could add an additional map to the manager for allDevices that is a map[string][]pluginapi.Device. This could be populated at the same time as healthy/unhealthy devices in the callback.

Pretty small addition that leaves the healthy/unhealthy devices untouched,